src/components/PopupWithForm.js

```diff
@@ -6,6 +6,7 @@ class PopupWithForm extends Popup {
     this._popupForm = this._popupElement.querySelector(".popup__form");
     this._handleFormSubmit = handleFormSubmit;
     this.inputs = this._popupForm.querySelectorAll("input");
+    this._submitButton = this._popupForm.querySelector('.popup__button');
   }
 
```

Comment:
Please, delete ` this._popupForm.reset();` from line 26 in `PopupWithForm`

You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.